app/services/album.js

```diff
@@ -1,27 +1,78 @@
 const axios = require('axios');
 const { url } = require('../../config').common.api;
 const errors = require('../errors');
+const { user: User } = require('../models');
+const { transaction: Transaction } = require('../models');
+const redisClient = require('redis').createClient();
+const DataLoader = require('dataloader');
+const RedisDataLoader = require('redis-dataloader')({ redis: redisClient });
 
-exports.getAlbum = id => {
-  const endpoint = `${url}albums/${id}`;
-  return axios
-    .get(endpoint)
-    .then(response => response.data)
-    .catch(e => {
-      throw errors.conectionError(e.message);
-    });
-};
+const albumLoader = new RedisDataLoader(
+  'album',
+  new DataLoader(
+    arrayId =>
+      Promise.resolve(
+        arrayId.map(id => {
+          const endpoint = `${url}albums/${id}`;
+          return axios.get(endpoint);
+        })
+      ),
+    { cache: false }
+  ),
+  {
+    // caching here is a local in memory cache. Caching is always done
+    // to redis.
+    cache: true,
+    serialize: info => JSON.stringify(info.data)
+  }
+);
 
-exports.getPhotosOfAlbum = albumId => {
-  const query = `?albumId=${albumId}`;
-  const endpoint = `${url}photos${query}`;
-  return axios
-    .get(endpoint)
-    .then(response => response.data)
-    .catch(e => {
-      throw errors.conectionError(e.message);
-    });
-};
+const allAlbumsLoader = new RedisDataLoader(
+  'allAlbums',
+  new DataLoader(arrayId =>
+    Promise.resolve(
+      arrayId.map(() => {
+        const endpoint = `${url}albums`;
+        return axios.get(endpoint);
+      })
+    )
+  ),
+  {
+    // caching here is a local in memory cache. Caching is always done
+    // to redis.
+    cache: true,
+    serialize: info => JSON.stringify(info.data)
```

Comment:
why do you need stringify?

Reply:
It is necessary because I just need stringify info.data